File: day-2/part-1.js
const data = require('./data')

let numValidPasswords = 0

data.forEach(datum => {
  const [minMax, charWithSuffix, word] = datum.split(' ')
  const [minChar, maxChar] = minMax.split('-')
  const min = Number(minChar)
  const max = Number(maxChar)
  const char = charWithSuffix.replace(':', '')

  let count = 0

  for (let i = 0; i < word.length; i++) {
    if (word[i] === char) {
      count += 1
    }
  }

  if (count >= min && count <= max) {
    numValidPasswords += 1
  }
})

console.log('numValidPasswords: ', numValidPasswords)

File: day-2/part-1.py
with open('data.txt') as f:
  data = f.read().splitlines()

  numValidPasswords = 0

  for datum in data:
    minMax, charWithSuffix, word = datum.split(' ')
    minChar, maxChar = minMax.split('-')
    min = int(minChar)
    max = int(maxChar)
    char = charWithSuffix.replace(':', '')

    count = 0

    for i in range(len(word)):
      if word[i] == char:
        count += 1
    
    if count >= min and count <= max:
      numValidPasswords += 1

  print(numValidPasswords)